tests/src/JIT/Performance/CodeQuality/HWIntrinsic/X86/PacketTracer/PacketTracer.cs

```diff
@@ -0,0 +1,253 @@
+        var isNull = Compare(isect.Distances, Intersections.NullDistance, FloatComparisonMode.EqualOrderedNonSignaling);
+        var backgroundColor = ColorPacket256Helper.BackgroundColor.Xs;
+        return new ColorPacket256(BlendVariable(color.Xs, backgroundColor, isNull),
+                                  BlendVariable(color.Ys, backgroundColor, isNull),
+                                  BlendVariable(color.Zs, backgroundColor, isNull));
+    }
+
+    private Vector256<float> TestRay(RayPacket256 rayPacket256, Scene scene)
+    {
+        var isect = MinIntersections(rayPacket256, scene);
+        if (isect.AllNullIntersections())
+        {
+            return SetZeroVector256<float>();
+        }
+        var isNull = Compare(isect.Distances, Intersections.NullDistance, FloatComparisonMode.EqualOrderedNonSignaling);
+        return BlendVariable(isect.Distances, SetZeroVector256<float>(), isNull);
+    }
+
+    private Intersections MinIntersections(RayPacket256 rayPacket256, Scene scene)
+    {
+        Intersections mins = new Intersections(Intersections.NullDistance, Intersections.NullIndex);
+        for (int i = 0; i < scene.Things.Length; i++)
+        {
+            Vector256<float> distance = scene.Things[i].ToPacket256().Intersect(rayPacket256);
+
+            if (!Intersections.AllNullIntersections(distance))
+            {
+                var notNullMask = Compare(distance, Intersections.NullDistance, FloatComparisonMode.NotEqualOrderedNonSignaling);
+                var nullMinMask = Compare(mins.Distances, Intersections.NullDistance, FloatComparisonMode.EqualOrderedNonSignaling);
+
+                var lessMinMask = Compare(mins.Distances, distance, FloatComparisonMode.GreaterThanOrderedNonSignaling);
+                var minMask = And(notNullMask, (Or(nullMinMask, lessMinMask)));
+                var minDis = BlendVariable(mins.Distances, distance, minMask);
+                var minIndeces = StaticCast<float, int>(BlendVariable(StaticCast<int, float>(mins.ThingIndeces),
+                                                                      StaticCast<int, float>(SetAllVector256<int>(i)),
+                                                                      minMask));
+                mins.Distances = minDis;
+                mins.ThingIndeces = minIndeces;
+            }
+        }
+        return mins;
+    }
+
+    private ColorPacket256 Shade(Intersections isect, RayPacket256 rays, Scene scene, int depth)
+    {
+
+        var ds = rays.Dirs;
+        var pos = isect.Distances * ds + rays.Starts;
+        var normals = scene.Normals(isect.ThingIndeces, pos);
+        var reflectDirs = ds - (Multiply(VectorPacket256.DotProduct(normals, ds), SetAllVector256<float>(2)) * normals);
+        var colors = GetNaturalColor(isect.ThingIndeces, pos, normals, reflectDirs, scene);
+
+        if (depth >= MaxDepth)
+        {
+            return colors + new ColorPacket256(.5f, .5f, .5f);
```

Comment:
Should there be an overload that takes a single float, and sets all elements to it?

Reply:
The constructor of `VectorPacket256` already uses `SetAllVector256` in this overload.

Reply:
Right, but I think that is still more expensive than an overload which takes a single float, calls `SetAllVector256` once, and then assigns all three `Vector256` fields.